### ISSUE TITLE
[OGRE-154] When a monitor is in portrait mode FSAA could not be set from ogre.cfg file on restoreConfig()

### ISSUE THREAD

**Issue by paroj** [paroj/ogre#34]
<b>[reporter="bjoe", created="Tue, 26 Feb 2013 14:52:46 +0100"]</b>
<p>When a monitor is set to portrait mode (in Windows 7), the FSAA  can accept only one value - '0' (NO Anti Alias).<br/>
Changing to a video mode with the right resolution, brings back the FSAA options (0, 2, 4, 8, 8 Quality)</p>

<p>The current implementation of restoreConfig() reads the ogre.cfg data to a MultiMap and then iterates through and sets the render system parameters. The problem is that it sets the 'FSAA' parameter BEFORE the 'Video Mode' parameter and causing the FSAA value to be ignored.</p>

<p>The obvious solution is to change the order of which the parameters are set so that the 'Video Mode' is set before the 'FSAA'.</p>

**Comment by paroj** [comment]
<b><i>[author="spacegaier", created="Sat, 28 Sep 2013 15:29:01 +0200"]</i></b>
<p>What render system is causing that issue?</p>

**Comment by paroj** [comment]
<b><i>[author="bjoe", created="Sat, 28 Sep 2013 15:36:40 +0200"]</i></b>
<p>I worked with D3D9.</p>

**Comment by paroj** [comment]
<b><i>[author="spacegaier", created="Sat, 28 Sep 2013 19:22:13 +0200"]</i></b>
<p>From my initial investigation, I think the issue is a bit different.</p>

<p>#1: It is by design, that only once a valid video mode has been set, possible FSAA values can be proposed (since only once we know the video mode the D3D9 rendering system provides those values).</p>

<p>#2: The problematic part is, that when in portrait mode the config dialog inits the video mode value with the default "800x600 32bit" which is invalid in that display orientation. That is something I will have to look into.</p>

<p>When you call the application for a second time in portrait mode, the correct video mode is set (if you selected one in the first run), but the old FSAA selection is not restored!</p>

**Comment by paroj** [comment]
<b><i>[author="bjoe", created="Sat, 28 Sep 2013 23:09:17 +0200"]</i></b>
<p>I think you are right, but this is another issue, there are two issues here:<br/>
1. Handling of the portrait mode (as you mentioned)<br/>
2. The parsing of the ogre.cfg file within the restoreConfig(), as I have described</p>
